### ISSUE TITLE
ValueError: Must pass either crs or epsg

### ISSUE THREAD

**Issue by sukruburakcetin** [opengeos/segment-geospatial#12]
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[11], line 2
      1 shapefile = 'segment_gpu_new.shp'
----> 2 sam.tiff_to_vector(mask, shapefile)

File ~\Desktop\segment-geospatial\samgeo\samgeo.py:172, in SamGeo.tiff_to_vector(self, tiff_path, output, simplify_tolerance, **kwargs)
    169         i["geometry"] = i["geometry"].simplify(tolerance=simplify_tolerance)
    171 gdf = gpd.GeoDataFrame.from_features(fc)
--> 172 gdf.set_crs(epsg=src.crs.to_epsg(), inplace=True)
    173 gdf.to_file(output, **kwargs)

File ~\Anaconda3\envs\segment-geospatial\lib\site-packages\geopandas\geodataframe.py:1279, in GeoDataFrame.set_crs(self, crs, epsg, inplace, allow_override)
   1277 else:
   1278     df = self
-> 1279 df.geometry = df.geometry.set_crs(
   1280     crs=crs, epsg=epsg, allow_override=allow_override, inplace=True
   1281 )
   1282 return df

File ~\Anaconda3\envs\segment-geospatial\lib\site-packages\geopandas\geoseries.py:1031, in GeoSeries.set_crs(self, crs, epsg, inplace, allow_override)
   1029     crs = CRS.from_epsg(epsg)
   1030 else:
-> 1031     raise ValueError("Must pass either crs or epsg.")
   1033 if not allow_override and self.crs is not None and not self.crs == crs:
   1034     raise ValueError(
   1035         "The GeoSeries already has a CRS which is not equal to the passed "
   1036         "CRS. Specify 'allow_override=True' to allow replacing the existing "
   1037         "CRS without doing any transformation. If you actually want to "
   1038         "transform the geometries, use 'GeoSeries.to_crs' instead."
   1039     )
ValueError: Must pass either crs or epsg.
```

Hello everyone, both the mask that is created and tiff have been projected with 4326, WGS 84 coordinate system. It can be easily seen on the map with their current projection without on the fly action. The mask that was created after sam generate function has 4326 coordinate system. However, when I try to use **tiff_to_vector** or **tiff_to_package** functions, I got no crs or epsg is set error. How to pass crs or epsg into these functions? 

I implemented the [notebook tutorial](https://samgeo.gishub.org/examples/satellite/) code with my projected tiff exactly as it is described.
Thanks in advance.

**Comment by sukruburakcetin** [comment]
![sam_geo1](https://user-images.githubusercontent.com/9219971/234367042-b70fa863-3b83-4899-8146-3dd648382286.png)

I used raster to polygon geoprocessing tool in order to see how good the result is overlayed.

![sam_geo2](https://user-images.githubusercontent.com/9219971/234367083-5bae2d1f-a3d3-468b-991f-b4d05c6fae3d.png)

This is the geospatial reference information screen that shows the mask that is created with the sam-generate wgs84 coordinate system.

![sam_geo3](https://user-images.githubusercontent.com/9219971/234367706-919cf91a-c585-4f12-87d7-714143ea6c2c.png)
And this is the original tiff file that is used for mask generation. It also has wgs84 coordinate system as described in its geospatial reference information.

By the way, I would like to say huge thanks for creating this amazing module open source, thank you so much for your amazing work. @giswqs  You are the real MVP in GIS.

I would like to say sorry if my workflow is not correct and flagged this as a bug. Regards.

**Comment by giswqs** [comment]
What `bbox` did you use?

**Comment by giswqs** [comment]
Can you provide complete source code that can reproduce the issue?

**Comment by sukruburakcetin** [comment]
I didn't pass any bbox value since I thought the clipped region that I derivated from the inclusive aerial image has already bbox regions inside of its metadata. 

I didn't use this:
```
if m.user_roi_bounds() is not None:
    bbox = m.user_roi_bounds()
else:
    bbox = [-95.3704, 29.6762, -95.368, 29.6775]
```
   
and this function:
`tms_to_geotiff(output=image, bbox=bbox, zoom=20, source='Satellite')`

This is my notebook:
![sam_geo4](https://user-images.githubusercontent.com/9219971/234371454-a6825789-d7be-4efb-adbb-83cd329fd2f4.png)

**Comment by giswqs** [comment]
Can you share the input geotiff for debugging?

**Comment by giswqs** [comment]
It appears that the inference runs successfully. You can use the following function directly to debug the issue. 

https://samgeo.gishub.org/samgeo/#samgeo.samgeo.SamGeo.tiff_to_vector
```python
  def tiff_to_vector(tiff_path, output, simplify_tolerance=None, **kwargs):
      """Convert a tiff file to a gpkg file.

      Args:
          tiff_path (str): The path to the tiff file.
          output (str): The path to the vector file.
          simplify_tolerance (_type_, optional): The simplify tolerance. Defaults to None.
      """

      with rasterio.open(tiff_path) as src:
          band = src.read()

          mask = band != 0
          shapes = features.shapes(band, mask=mask, transform=src.transform)

      fc = [
          {"geometry": shapely.geometry.shape(shape), "properties": {"value": value}}
          for shape, value in shapes
      ]
      if simplify_tolerance is not None:
          for i in fc:
              i["geometry"] = i["geometry"].simplify(tolerance=simplify_tolerance)

      gdf = gpd.GeoDataFrame.from_features(fc)
      gdf.set_crs(epsg=src.crs.to_epsg(), inplace=True)
      gdf.to_file(output, **kwargs)

**Comment by giswqs** [comment]
You can upload it to me via this [link](https://www.dropbox.com/request/lMU5exDZ4rTLQEu8rScH).

**Comment by sukruburakcetin** [comment]
Okay, with the inference function you have provided, I was able to create a shape vector file with WKID code 3857 without any complications(it is a different projection so you cannot overlay it properly). However, when I try it with the code 4326, the shapefile was created with polygon attributes but you cannot show data on the map viewer. 

I have tried a bunch of other epsg codes, so you can easily see output on the viewer but it cannot be overlayed on the real tiff data due to their projections do not match. They all work but in a way of is not expected. 

This is how I debugged it:
![sam_geo6](https://user-images.githubusercontent.com/9219971/234385881-a9160b67-5352-4e3b-b35b-689f89d7e262.png)
And this is the output:
![sam_geo5](https://user-images.githubusercontent.com/9219971/234385936-f2bf559c-99be-4d8b-8d5e-90d09b9dc2e4.png)

All of them work but their end results don't match(it is expected). The only one which is not worked: is the attempt with code 4326. The shapefile was created but cannot be seen.

Do you mind if I request to try it out with the sample I provided via your link? Thank you so much for your patience and efforts, dear Wu.

**Comment by giswqs** [comment]
I think the problem is your input dataset. From the image extent numbers, it is clearly those numbers are not lat/lon. They are likely epsg:3857. But the CRS shows that it is WGS 1984, which is wrong. So my question is how did you get that dataset? The CRS was not assigned properly. 

![image](https://user-images.githubusercontent.com/5016453/234402276-918f7c10-1bbb-4f65-9569-1d79f0f16a5c.png)

![image](https://user-images.githubusercontent.com/5016453/234402378-75382824-4061-471f-9745-9b5e5870fbda.png)

The correct one should look like this.
![image](https://user-images.githubusercontent.com/5016453/234403061-a8f5c2d5-3c73-4cdc-a5fb-e06868693ef5.png)

**Comment by sukruburakcetin** [comment]
Thank you so much, dear Wu. 

The problem is solved by changing the data's geospatial coordinate reference system into GCS_WGS_1984(4326) from pseudo-Mercator(3857), so extend parameters are converted into decimal degrees and the inference code worked flawlessly.
![sam_geo7](https://user-images.githubusercontent.com/9219971/234508028-87a43eec-9c6a-47d5-9c5c-e54948e1c72a.png)

It is right that there is a problem with the main sample tiff, I derived it from our database(which is shown and used in [here](https://sehirharitasi.ibb.gov.tr/ as "2022 Uydu" in the layer selection section). The sample tiff was ESRGAN applied version of raw data(It was the project in which I enhanced raw data with super-resolution techniques). There could be an issue when mosaicking tile files into a raster file.
![sam_geo8](https://user-images.githubusercontent.com/9219971/234507897-76c195ed-e6ee-4ff3-a3f7-009ae008908f.png)

However, bear in mind: The function in the sample notebook still asks for crs for being set by the user while the inference function runs great if you set the crs in it.
```
shapefile = 'segment_gpu_new_decimal_degrees.shp'
sam.tiff_to_vector(mask, shapefile)
```
